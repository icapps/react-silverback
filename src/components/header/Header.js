import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { strings } from '../../utils';
import './header.css';
import constants from '../../redux/auth/constants';
import Button from '../button/Button';

const logout = require('../../assets/images/logout.svg');

class Header extends React.Component {
  handleLogout = async () => {
    const result = await this.props.logout();
    if (result.action && (result.action.type === constants.LOGOUT_USER_FULFILLED || result.action.type === constants.LOGOUT_USER_REJECTED)) {
      window.location.href = '/';
    }
  }

  render() {
    const { props } = this;

    return (
      <header className="navbar navbar-dark bg-dark fixed-top">
        <Link className="navbar-brand" to="/">{strings.HEADER_TITLE}</Link>
        <div>
          <Button
            rightIcon={logout}
            text={strings.LOGOUT}
            className='btn-dark btn-logout'
            handleClick={this.handleLogout}
          />
          <span className="navbar-toggler-icon" onClick={props.toggleNavigation} />
        </div>
      </header>
    );
  }
};

Header.propTypes = {
  toggleNavigation: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
};

export default Header;
