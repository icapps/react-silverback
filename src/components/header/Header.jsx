import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { strings } from '../../utils';
import './header.css';

const Header = props => (
  <header className="navbar navbar-dark bg-dark fixed-top">
    <Link className="navbar-brand" to="/">{strings.HEADER_TITLE}</Link>
    <span className="navbar-toggler-icon" onClick={props.toggleNavigation} />
  </header>
);

Header.propTypes = {
  toggleNavigation: PropTypes.func.isRequired,
};

export default Header;
