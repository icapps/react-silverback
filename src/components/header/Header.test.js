import React from 'react';
import { shallow, configure } from 'enzyme';
import Header from './Header';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });

describe('Header Component', () => {
  it('should render a Header component', () => {
    const wrapper = shallow(<Header toggleNavigation={() => { }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
