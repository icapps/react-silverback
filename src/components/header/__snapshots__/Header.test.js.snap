// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render a Header component 1`] = `ShallowWrapper {}`;
